pkg/obfuscator/domain_test.go

```diff
@@ -3,106 +3,61 @@ package obfuscator
+				"received request on https://docs.okd.io",
 			},
 			output: []string{
-				"received request on obfuscated0001.ext",
-				"received request on https://obfuscated0001.ext",
-			},
-			report: map[string]string{"test.com": "obfuscated0001.ext"},
-		},
-		{
-			name: "subdomain",
-			tlds: []string{".com", ".info", ".org"},
-			input: []string{
-				"received request on abc.test.com",
-				"received request on def.test.com",
-			},
-			output: []string{
-				"received request on abc.obfuscated0001.ext",
-				"received request on def.obfuscated0001.ext",
+				"received request on domain0000001",
+				"received request on https://docs.domain0000002",
 			},
 			report: map[string]string{
-				"abc.test.com": "abc.obfuscated0001.ext",
-				"def.test.com": "def.obfuscated0001.ext",
+				"openshift.com": "domain0000001",
+				"docs.okd.io":   "docs.domain0000002",
 			},
 		},
 		{
-			name: "multiple domains",
-			tlds: []string{".com", ".info", ".org"},
+			name:    "subdomains",
+			domains: []string{"test.com", "test.info", "test.org"},
 			input: []string{
-				"request to test1.com failed",
-				"request to test2.com succeeded",
-			},
-			output: []string{
-				"request to obfuscated0001.ext failed",
-				"request to obfuscated0002.ext succeeded",
-			},
-			report: map[string]string{
-				"test1.com": "obfuscated0001.ext",
-				"test2.com": "obfuscated0002.ext",
-			},
-		},
-		{
-			name: "non tld host",
-			tlds: []string{".com", ".info", ".org"},
-			input: []string{
-				"request to test1.remotehost failed",
-				"request to test2.remotehost succeeded",
-			},
-			output: []string{
-				"request to test1.obfuscated0001 failed",
-				"request to test2.obfuscated0001 succeeded",
-			},
-			report: map[string]string{
-				"test1.remotehost": "test1.obfuscated0001",
-				"test2.remotehost": "test2.obfuscated0001",
-			},
-		},
-		{
-			name: "ignore ip addresses",
-			input: []string{
-				"calling 192.168.1.10 for backlog",
-				"calling http://192.168.1.10 for backlog",
-			},
-			output: []string{
-				"calling 192.168.1.10 for backlog",
-				"calling http://192.168.1.10 for backlog",
-			},
-			report: map[string]string{},
-		},
-		{
-			name: "ip address and host",
-			tlds: []string{".com"},
-			input: []string{
-				"received request for https://service.example.com from 192.168.1.20",
+				"received request on abc.test.com",
+				"received request on def.test.info",
+				"received request on ghi.abc.test.com",
+				"received request on pqr.ghi.abc.test.com",
 			},
 			output: []string{
-				"received request for https://service.obfuscated0001.ext from 192.168.1.20",
+				"received request on abc.domain0000001",
+				"received request on def.domain0000002",
+				"received request on ghi.abc.domain0000001",
+				"received request on pqr.ghi.abc.domain0000001",
 			},
 			report: map[string]string{
-				"service.example.com": "service.obfuscated0001.ext",
+				"abc.test.com":         "abc.domain0000001",
```

Comment:
can you also add a test around those subdomains? 
so what if I wanted to have `abc.test.com` obfuscated, instead of just the `test.com`.

Reply:
Added a test